src/json-decoder.ts

```diff
@@ -494,6 +494,69 @@ export namespace JsonDecoder {
       }
     });
   }
+
+  /**
+   * Transforms union type to intersection type.
+   *
+   * For example:
+   *
+   *    Intersect<{ x: number } | { y: number }>
+   *
+   * becomes
+   *
+   *    { x: number } & { y: number }
+   */
+  export type Intersect<T> = (T extends any ? (x: T) => void : never) extends (
+    x: infer R
+  ) => void
+    ? R
+    : never;
+
+  /**
+   * Transforms array of objects to a combined intersection type
+   *
+   * For example:
+   *
+   *    Combine<[{ x: number }, { y: number }]>
+   *
+   * becomes
+   *
+   *    { x: number } & { y: number }
+   */
+  export type Combine<T extends { [k: string]: any }[]> = Intersect<T[number]>;
```

Comment:
I don't think we need to export the `Combine` and `Intersect` types. Is it required for the feature to work?

Reply:
It should work without exports. I think it's just a habit to export every type definition 😅 Updated now.

Reply:
Yeah, I normally do that as well!
In case of an open-source library I guess I prefer to keep the API surface as small as possible.
Thanks!